crates/miden-tx/src/executor/exec_host.rs

```diff
@@ -166,10 +137,41 @@ where
 
     /// Handles the [`TransactionEvent::EpilogueTxFeeComputed`] and returns an error if the account
     /// cannot pay the fee.
-    fn on_tx_fee_computed(
+    async fn on_tx_fee_computed(
         &self,
         fee_asset: FungibleAsset,
     ) -> Result<Vec<AdviceMutation>, TransactionKernelError> {
+        let asset_witness = self
+            .base_host
+            .store()
+            .get_vault_asset_witness(
+                self.base_host.native_account_header().id(),
+                self.base_host.native_account_header().vault_root(),
+                fee_asset.vault_key(),
+            )
+            .await
+            .map_err(|err| TransactionKernelError::GetVaultAssetWitness {
+                vault_root: self.base_host.native_account_header().vault_root(),
+                vault_key: fee_asset.vault_key(),
+                source: Box::new(err),
+            })?;
+
+        // Find asset in the witness or default to 0 if it isn't present.
+        let initial_native_asset = asset_witness
```

Comment:
nit: while this is true now that the fee asset is a native asset, in the future this may not be the case. So, maybe a more future-proof name here would be `initial_fee_asset`?